frontend/src/metabase/visualizations/visualizations/SmartScalar/compute.unit.spec.js

```diff
@@ -1952,6 +1952,55 @@ describe("SmartScalar > compute", () => {
         },
       );
     });
+
+    describe("format options", () => {
+      const series = [
+        createMockSingleSeries(
+          {},
+          {
+            data: {
+              rows: [
+                [100, "2023-10-01T00:00:00"],
+                [300, "2024-10-01T00:00:00"],
+              ],
+              cols: [
+                createMockNumberColumn({ name: "Count" }),
+                createMockDateTimeColumn({ name: "Month" }),
+              ],
+            },
+          },
+        ),
+      ];
+
+      const insights = [{ unit: "month", col: "Count" }];
+
+      const createVizSettings = settings =>
+        createMockVisualizationSettings({
+          "scalar.field": "Count",
+          "scalar.comparisons": { type: COMPARISON_TYPES.PREVIOUS_VALUE },
+          ...settings,
+        });
+
+      it("should have `compact: false` by default", () => {
+        const { formatOptions } = computeTrend(
+          series,
+          insights,
+          createVizSettings(),
+        );
+        expect(formatOptions.compact).toBeFalsy();
+      });
+
+      it("should have `compact: true` with `scalar.compact_primary_number` viz setting", () => {
+        const { formatOptions } = computeTrend(
+          series,
+          insights,
+          createVizSettings({
+            "scalar.compact_primary_number": true,
+          }),
+        );
+        expect(formatOptions.compact).toBe(true);
+      });
```

Comment:
Can we add a check for the display values returned by `computeTrend`, since these display values will also be used by the static-viz and we want to insure this behaviors works across any future changes?

**Data Setup**
```javascript
const series = [
        createMockSingleSeries(
          {},
          {
            data: {
              rows: [
                [105000, "2023-10-01T00:00:00"],
                [210000, "2024-10-01T00:00:00"],
              ],
              cols: [
                createMockNumberColumn({ name: "Count" }),
                createMockDateTimeColumn({ name: "Month" }),
              ],
            },
          },
        ),
      ];
```

**Compact Setting is Off**
```javascript
const { comparison, display, formatOptions } = computeTrend(
          series,
          insights,
          createVizSettings(),
        );

const { display: comparisonDisplay } = comparison;

expect(formatOptions.compact).toBeFalsy();
expect(display.value).toBe("210,000")
expect(comparisonDisplay).toBe("105,000")
```

**Compact Setting is On**
```javascript
const { comparison, display, formatOptions } = computeTrend(
          series,
          insights,
          createVizSettings({
            "scalar.compact_primary_number": true,
          }),
        );

const { display: comparisonDisplay } = comparison;

expect(formatOptions.compact).toBe(true);
expect(display.value).toBe("210k")
expect(comparisonDisplay).toBe("105k")
```